test/common/filter/auth/client_ssl_test.cc

```diff
@@ -149,8 +149,10 @@ TEST_F(ClientSslAuthFilterTest, Ssl) {
   // Create a new filter for an SSL connection, with no backing auth data yet.
   createAuthFilter();
   ON_CALL(filter_callbacks_.connection_, ssl()).WillByDefault(Return(&ssl_));
-  Network::Address::Ipv4Instance remote_address("192.168.1.1");
-  EXPECT_CALL(filter_callbacks_.connection_, remoteAddress()).WillOnce(ReturnRef(remote_address));
+  Network::Address::InstanceConstSharedPtr remote_address =
+      Network::Utility::parseInternetAddress("192.168.1.1");
+  EXPECT_CALL(filter_callbacks_.connection_, remoteAddressConstSharedPtr())
+      .WillOnce(Return(remote_address));
```

Comment:
I remember in a previous PR, I ran into some problems with testing::Return(smart pointer). Here, it seems fine I'm assuming because it's constant. @htuch can you take a second look?